imap_processing/cli.py

```diff
@@ -517,7 +517,7 @@ def post_processing(
         self,
         processed_data: list[xr.Dataset | Path],
         dependencies: ProcessingInputCollection,
-    ) -> None:
+    ) -> Optional[list[Path]]:
```

Comment:
I don't love returning different values here. Could you change this to always return the list. I think if you move up `products = []` to the top, then in the early return case you can just `return products` there as well which would be an empty list.
```suggestion
    ) -> list[Path]:
```

Reply:
Thanks for the callout! We've updated this and added corresponding tests in the second commit.

Ethan and George